app/com/yetu/oauth2provider/controllers/setup/SetupController.scala

```diff
@@ -27,6 +33,58 @@ class SetupController(override implicit val env: RuntimeEnvironment[YetuUser]) e
     }
   }
 
+  override def handleSignUp(token: String) = {
+    Action.async {
+      implicit request =>
+        executeForToken(token, true, {
+          t =>
+            val newUser = YetuUser(
+              UUIDGenerator.uuid(),
+              providerId,
+              firstName = t.registrationInfo.map(info => info.firstName),
+              lastName = t.registrationInfo.map(info => info.lastName),
+              fullName = t.registrationInfo.map(info => "%s %s".format(info.firstName, info.lastName)),
+              email = Some(t.email),
+              avatarUrl = None,
+              authMethod = AuthenticationMethod.UserPassword,
+              passwordInfo = t.registrationInfo.map(info => env.currentHasher.hash(info.password)),
+              userAgreement = t.registrationInfo.map(info => info.userAgreement),
+              registrationDate = Some(DateTime.now)
+            )
+
+            val withAvatar = env.avatarService.map {
```

Comment:
This code came from inside securesocial.. but we make a little but better I guess now:

```
val withAvatar = env.avatarService match {
              case Some(avatar) =>
                avatar.urlFor(t.email).map { url =>
                  if (url != newUser.avatarUrl) newUser.copyUser(avatarUrl = url) else newUser
                }
              case _ => Future.successful(newUser)
            }
```